### ISSUE TITLE
Rule Request: `allowed_prefix_symbols` configuration in `identifier_name`

### ISSUE THREAD

**Issue by philtre** [realm/SwiftLint#3682]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

When `validates_start_with_lowercase` is `true`, all identifier names must start with a lowercase character.
This will trigger a violation even if the name starts with a symbol, for example and underscore `_`.

The only way to achieve this now is to disable `validates_start_with_lowercase`, but it would be nice to have to option to allow certain prefix symbols, eg:

```
identifier_name:
  allowed_prefix_symbols: ["_"]
  validates_start_with_lowercase: true
```

The above **would** trigger a violation for:

```swift
var MyVar: String
```

...but **would not** trigger a violation for:

```swift
private var _myVar: String
```

While naming properties this way shouldn't be the norm, sometimes it can serve as raw value storage for a computed property of the same name. For example:

```swift
private var _myVar: Int?

@objc
public var myVar: Int {
    get { 
         _myVar ?? -1
    }
    set { 
        _myVar = newValue < 0 ? nil : newValue
    }
}
```
